### PR TITLE
fix: set active database from query

### PR DESCRIPTION
**Pull request by alespour**
Fixes https://github.com/influxdata/EAR/issues/5278

For InfluxQL metaqueries to work with Cloud Serverless, active database needs to be set (via `db` HTTP query parameter). Current version only supports settings active via `USE` clause.

This PR adds sets active database using value of  `ON` clause if present in the query and no db was set with `USE`.

  - [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: server/influx_test.go [hunk 1]
```diff
@@ -217,6 +217,104 @@ func TestService_Influx_UseCommand(t *testing.T) {
...
+			db:   "anotherdb",
+		},
+		{
+			name: `show tag keys on "mydb"`,
+			db:   "mydb",
+		},
+		{
+			name: `show tag keys on "mydb" from "table"`,
+			db:   "mydb",
+		},
```
Comment (bednar) [inline server/influx_test.go:252]: Please also add tests for the following edge cases:

```suggestion
		},
		{
			name: `show tag keys on "my_db" from "table"`,
			db:   "my_db",
		},
		{
			name: `show tag keys on "my-db" from "table"`,
			db:   "my-db",
		},
		{
			name: `show tag keys on "my/db" from "table"`,
			db:   "my/db",
		},
		{
			name: `show tag keys on "my db" from "table"`,
			db:   "my db",
		},
```

**Review by bednar — CHANGES_REQUESTED** [review]
Thank you for your PR! 👍 Before we can proceed with accepting it, there are a few requirements that need to be met:

The `InfluxDB v1` allows the creation of database names with spaces. We should also cover this use case:

image

File: server/influx_test.go [hunk 2]
```diff
@@ -217,6 +217,104 @@ func TestService_Influx_UseCommand(t *testing.T) {
...
+		{
+			name: "USE anotherdb; SHOW TAG KEYS ON mydb",
+			db:   "anotherdb",
+		},
+		{
+			name: `show tag keys on "mydb"`,
+			db:   "mydb",
+		},
```
Comment (bednar) [inline server/influx_test.go:248]: We should also cover the typo error syntax such as: `oN`:

```suggestion
		{
			name: `show tag keys on "mydb"`,
			db:   "mydb",
		},
		{
			name: `show tag keys oN "mydb"`,
			db:   "mydb",
		},
```

**Review by bednar — CHANGES_REQUESTED** [review]

File: server/influx.go [hunk 3]
```diff
@@ -173,26 +173,50 @@ func (s *Service) Write(w http.ResponseWriter, r *http.Request) {
...
+
+	// allow to set active database with USE command or via ON clause, examples:
 	//  use mydb
 	//  use "mydb"
 	//  USE "mydb"."myrp"
 	//  use "mydb.myrp"
 	//  use mydb.myrp
+	//  show tag keys on "mydb"
+	//  SHOW TAG KEYS ON "mydb"
 	if strings.HasPrefix(req.Command, "use ") || strings.HasPrefix(req.Command, "USE ") {
```
Comment (bednar) [inline server/influx.go:204]: To be safe with lower and upper cases, we can use something like:

```suggestion
	//  SHOW TAG KEYS ON "mydb"
	command := strings.ToLower(req.Command)
	if strings.HasPrefix(command, "use ") {
```

File: server/influx.go [hunk 4]
```diff
@@ -173,26 +173,50 @@ func (s *Service) Write(w http.ResponseWriter, r *http.Request) {
...
-					req.RP = dbrp[1]
-				}
+			if useDb(dbSpec) == nil {
 				req.Command = strings.TrimSpace(req.Command[nextCommand+1:])
 			}
 		}
+	} else if strings.Contains(req.Command, " on ") || strings.Contains(req.Command, " ON ") {
+		fields := strings.Fields(req.Command)
+		for i, field := range fields {
+			if field == "on" || field == "ON" {
```
Comment (bednar) [inline server/influx.go:214]: To be safe with lower and upper cases, we can use something like:

```suggestion
	} else if strings.Contains(command, " on ") {
		fields := strings.Fields(req.Command)
		for i, field := range fields {
			if strings.ToLower(field) == "on" {
```

**Review by bednar — CHANGES_REQUESTED** [review]

**Review by bednar — APPROVED** [review]
LGTM 🚀
